Src/VimWpf/Implementation/ImeCoordinator/ImeCoordinator.cs

```diff
@@ -0,0 +1,423 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Windows.Input;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Vim.UI.Wpf.Implementation.ImeCoordinator
+{
+    [Export(typeof(IVimBufferCreationListener))]
+    internal sealed class ImeCoordinator : IVimBufferCreationListener, IDisposable

```

Comment:
Is this `IDisposable` invoked during normal VS operation? I don't believe the Vim implementation will probe `IVimBufferCreationListener` for `IDisposable`.

Reply:
I gathered from the MEF comments elsewhere in VsVim that if MEF creates it and it's `IDisposable`, that MEF will dispose it. It's not particularly critical as this is basically a singleton for the duration of Visual Studio like VsVim itself.